### PR TITLE
Oakfieldset component

### PR DESCRIPTION
**Pull request by k-huggs**
# How to review this PR

_Leave this text block for the reviewer_

- Check [component hierarchy](https://miro.com/app/board/uXjVNnKBgyk=/?share_link_id=59445593794) is followed correctly
- Check the design [Heuristics](https://components.thenational.academy/?path=/docs/docs-howtodesigncomponents--docs#heuristics-for-component-design) have been followed
- Check [naming conventions](https://components.thenational.academy/?path=/docs/docs-namingconventions--docs) have been applied
- Check for these gotchyas:
  - Missing exports for Oak components
  - Accidental export of Internal components
  - Snapshots of unexpected components have been modified
  - Circular dependencies
  - Code duplication (via not using base components)
  - Non-functional storybook for this or related components
  - Sensitve files changed eg. atoms, or style tokens
  - Relative imports
  - Default exports

# Add your PR description below

### CODE REVIEW

File: src/components/atoms/OakFieldset/OakFieldset.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  margin: 0;
+  padding: 0;
+  ${colorStyle}
+  ${sizeStyle}
+  ${opacityStyle}
+  ${marginStyle}
+  ${paddingStyle}
+  ${borderStyle}
+  ${typographyStyle}
+`;
```
Comment (carlmw) [inline src/components/atoms/OakFieldset/OakFieldset.tsx:41]: I think this should have the same props as `OakBox` if you look at `OakForm` it imports the styles and props from `OakBox` so you don't have to compose all the styles yourself.

Something like this 👇🏻 
```suggestion
import styled from "styled-components";

import { OakBoxProps, oakBoxCss } from "@/components/atoms/OakBox";

export type OakFieldsetProps = OakBoxProps;
/**
 * OakFieldset renders a custom `fieldset` component, removes default styling of fieldset.
  color, opacity, margin, padding, border and typography styles can be passed in also.
 */
export const OakFieldset = styled.fieldset<OakFieldsetProps>`
  border: 0;
  margin: 0;
  padding: 0;
  ${oakBoxCss};
`;

```

**Review by carlmw — COMMENTED** [review]
This should probably expose the same props as `OakBox` so i've made that suggestion

File: src/components/atoms/OakFieldset/OakFieldset.tsx (hunk 1 above)
Comment (k-huggs) [reply src/components/atoms/OakFieldset/OakFieldset.tsx:41]: Not sure if you've watched The Bear (recommended if not)

Comment (carlmw) [reply src/components/atoms/OakFieldset/OakFieldset.tsx:41]: Not sure if you've played Overcooked https://www.tiktok.com/@mitchell_ferguson/video/7324905711157202218

**Review by carlmw — APPROVED** [review]
👨🏻‍🍳

**Pull request merged**